sdk/daml-lf/interpreter/src/test/scala/com/digitalasset/daml/lf/speedy/ExceptionTest.scala

```diff
@@ -720,6 +768,346 @@ class ExceptionTest(majorLanguageVersion: LanguageMajorVersion)
 
   }
 
+  // Section testing exceptions thrown when computing the metadata of a contract
+  {
+    val parserParameters =
+      defaultParserParameters.copy(
+        // TODO: revert to the default version once it supports upgrades
+        languageVersion = LanguageVersion.Features.packageUpgrades
+      )
+
+    // A package that defines an interface, a key type, an exception, and a party to be used by
+    // the packages defined below.
+    val commonDefsPkgId = Ref.PackageId.assertFromString("-common-defs-v1-")
+    val commonDefsPkg =
+      p"""metadata ( '-common-defs-' : '1.0.0' )
+          module Mod {
+            record @serializable MyUnit = {};
+            interface (this : Iface) = { viewtype Mod:MyUnit; };
+
+            record @serializable Key = { label: Text, maintainers: List Party };
+
+            record @serializable Ex = { message: Text } ;
+            exception Ex = {
+              message \(e: Mod:Ex) -> Mod:Ex {message} e
+            };
+
+          val mkParty : Text -> Party = \(t:Text) -> case TEXT_TO_PARTY t of None -> ERROR @Party "none" | Some x -> x;
+          val alice : Party = Mod:mkParty "alice";
+          }
+      """ (parserParameters.copy(defaultPackageId = commonDefsPkgId))
+
+    /** An abstract class whose [[templateDefinition]] method generates LF code that defines a template named
+      * [[templateName]].
+      * The class is meant to be extended by concrete case objects which override one the metadata's expressions with an
+      * expression that throws an exception.
+      */
+    abstract class TemplateGenerator(val templateName: String) {
+      def precondition = """True"""
+      def signatories = s"""Cons @Party [Mod:${templateName} {p} this] (Nil @Party)"""
+      def observers = """Nil @Party"""
+      def agreement = """"agreement""""
+      def key =
+        s"""
+           |  '$commonDefsPkgId':Mod:Key {
+           |    label = "test-key",
+           |    maintainers = (Cons @Party [Mod:${templateName} {p} this] (Nil @Party))
+           |  }""".stripMargin
+
+      def maintainers =
+        s"""\\(key: '$commonDefsPkgId':Mod:Key) -> ('$commonDefsPkgId':Mod:Key {maintainers} key)"""
```

Comment:
those escape are a bit annoying, but I assume there is any simple solution to avoid them.